src/Stack/Hoogle.hs

```diff
@@ -75,49 +80,64 @@ hoogleCmd (args,setup,rebuild,startServer) =
-
-          -- FIXME For a while, we've been following the logic of
-          -- taking the latest Hoogle version available. However, we
-          -- may want to instead grab the version of Hoogle present in
-          -- the snapshot current being used instead.
-          pure $ fromMaybe (Left hoogleMinIdent) $ do
-            pir@(PackageIdentifierRevision _ ver _) <- mversion
-            guard $ ver >= hoogleMinVersion
-            Just $ Right pir
-
-        case hooglePackageIdentifier of
-            Left{} -> logInfo $
-              "Minimum " <>
-              fromString (packageIdentifierString hoogleMinIdent) <>
-              " is not in your index. Installing the minimum version."
-            Right ident -> logInfo $
-              "Minimum version is " <>
-              fromString (packageIdentifierString hoogleMinIdent) <>
-              ". Found acceptable " <>
-              display ident <>
-              " in your index, installing it."
+    requiringHoogle :: Muted -> RIO EnvConfig x -> RIO EnvConfig x
+    requiringHoogle muted f = do
+        hoogleTarget <- do
+          sourceMap <- view $ sourceMapL . to smDeps
+          case Map.lookup hooglePackageName sourceMap of
+            Just hoogleDep ->
+              case dpLocation hoogleDep of
+                PLImmutable pl -> do
+                  T.pack . packageIdentifierString <$>
+                      restrictMinHoogleVersion muted (packageLocationIdent pl)
+                PLMutable _ -> do
+                  muteableLogInfo muted "Local hoogle package found, installing it"
+                  pure $ T.pack (packageNameString hooglePackageName)
+            Nothing -> do
+              mpir <- getLatestHackageVersion YesRequireHackageIndex hooglePackageName UsePreferredVersions
+              let hoogleIdent = case mpir of
+                      Nothing -> hoogleMinIdent
+                      Just (PackageIdentifierRevision _ ver _) ->
+                          PackageIdentifier hooglePackageName ver
+              T.pack . packageIdentifierString <$>
+                  restrictMinHoogleVersion muted hoogleIdent
         config <- view configL
         let boptsCLI = defaultBuildOptsCLI
-                { boptsCLITargets =
-                    pure $
-                    T.pack . packageIdentifierString $
-                    either
-                    id
-                    (\(PackageIdentifierRevision n v _) -> PackageIdentifier n v)
-                    hooglePackageIdentifier
+                { boptsCLITargets =  [hoogleTarget]
                 }
         runRIO config $ withEnvConfig NeedTargets boptsCLI f
+    restrictMinHoogleVersion
+      :: HasLogFunc env
+      => Muted -> PackageIdentifier -> RIO env PackageIdentifier
+    restrictMinHoogleVersion muted ident = do
+      if ident < hoogleMinIdent
+      then do
+          muteableLogInfo muted $
```

Comment:
I think a warning may make more sense here too: this is a situation that's likely to result in failure.